polaris/evaluate/_metric.py

```diff
@@ -45,6 +70,7 @@ class MetricInfo(BaseModel):
     is_multitask: bool = False
     kwargs: dict = Field(default_factory=dict)
     direction: DirectionType
+    needs_probs: bool = False
```

Comment:
In line with scikit-learn, I would like to use named arguments that are specific to what's expected, e.g. `def metric_1(y_true, y_pred)` vs `def metric_2(y_true, y_prob)`. I find that to be more clear and like to be consistent with Scikit-learn.

It seems scikit-learn uses `y_pred`, `y_prob` and `y_score`. What would you think of: 
```suggestion
    y_type: Literal["y_pred", "y_prob", "y_score"] = "y_pred"
```
?

 Then in `Metric.score()` we can do something like: 
```python
def score(
    self, 
    y_true: np.ndarray, 
    y_pred: Optional[np.ndarray] = None, 
    y_score: Optional[np.ndarray] = None, 
    y_prob: Optional[np.ndarray] = None
) -> float:
        
    if y_pred is None and y_score is None and y_prob is None: 
        raise ValueError(...)

    if self.y_type == "y_pred":
        pred = y_pred
    elif self.y_type == "y_score":
        pred = y_score
    else:
        pred = y_prob

    kwargs = {"y_true": y_true, self.y_type: pred}
    return self.fn(**kwargs)
```

I think it's more robust, but night complicate the API for the user, because you would probably have to propagate such changes to the API of the benchmark too.

What do you think?

Reply:
It would work. But isn't it additional work that user who defines metrics have to figure out the "y_type" for each metric?

Reply:
Maybe use `metric_args = inspect.signature(self.fn).parameters` to get the `y_type` instead.

Reply:
> It would work. But isn't it additional work that user who defines metrics have to figure out the "y_type" for each metric?

oh. I see. It's defined in the MetricInfo.

Reply:
The sklearn metrics used various argument names for `y_pred`. https://scikit-learn.org/stable/modules/model_evaluation.html#classification-metrics
Which is fine because they will be defined internally with MetricInfo. 
I am not sure whether to differenciate "y_score" and "y_prob". To me, one of them would be enough. 
In Benchmark.evaluate(), I think we can simplify them to either "y_score" or "y_prob". Maybe "y_score" is more universal.

Reply:
> In Benchmark.evaluate(), I think we can simplify them to either "y_score" or "y_prob".

Yes, agreed, but let's go with `y_prob` as you have it now.

From the scikit-learn docs on `y_score` (see also [here](https://scikit-learn.org/stable/glossary.html#term-decision_function)): 
> These can be either probability estimates or non-thresholded decision values

It seems that if we expect `y_score`, we can always provide `y_prob`, but not the other way around. I thus think `y_prob` will be the better choice and I also think its naming is better understood.